components/chart/src/chart.ts

```diff
@@ -207,10 +207,22 @@ export class DtChart
   implements AfterViewInit, OnDestroy, OnChanges, AfterContentInit {
   /** Options to configure the chart. */
   @Input()
-  get options(): DtChartOptions {
+  get options(): Observable<DtChartOptions> | DtChartOptions {
     return this._options;
   }
-  set options(options: DtChartOptions) {
+  set options(options: Observable<DtChartOptions> | DtChartOptions) {
+    if (this._optionsSub) {
```

Comment:
I'm not sure if I like the Idea of subscribing inside the setter. Maybe an approach with a BehaviourSubject that holds the value would be cleaner?

```typescript
class MyComp {

  // the internal stream that holds the value
  private _internalOptions$ = new BehaviorSubject<Observable<DtChartOptions> | DtChartOptions>(defaultOptions);
  // the options stream where should be always a stream then.
  options$ = this._internalOptions$.pipe(
    switchMap(value => (value instanceof Observable ? value : of(value)))
  );

  @Input()
  get options(): Observable<DtChartOptions> | DtChartOptions {
    return this._internalOptions$.value;
  }
  set options(options: Observable<DtChartOptions> | DtChartOptions) {
      this._internalOptions$.next(options);
  }
}
```

Reply:
@lukasholzer we are doing exactly the same logic that has the same result in the setter for the chart series. I would opt for consistency here.

Reply:
I would also keep it for consistency, I'm not a big fan of managing the subscription inside the setter but neither of using the subject `value` as it loses the sense of an observer itself.
Something it could be done, in further changes, is to adapt the whole update flow to react properly to the options and series changes

```ts
  combineLatest(
    this._options$,
    this._series$
  ).pipe(
    tap(update),
  );
```
and this could be done by wrapping options and series like @lukasholzer pointed:
```ts
  private _options$ = new BehaviorSubject<Observable<DtChartOptions> | DtChartOptions>(defaultOptions);
  private _options$ = new BehaviorSubject<Observable<DtChartSeries[]> | DtChartSeries[]> | undefined>(undefined);
```